ignite/base/mixins.py

```diff
@@ -22,3 +28,262 @@ def load_state_dict(self, state_dict: Mapping) -> None:
         opts = [k in state_dict for k in self._state_dict_one_of_opt_keys]
         if len(opts) > 0 and ((not any(opts)) or (all(opts))):
             raise ValueError(f"state_dict should contain only one of '{self._state_dict_one_of_opt_keys}' keys")
+
+
+class EventsDriven:
```

Comment:
Even if it’s a mixin, it’s an important library’s feature, as Metric for instance. It would deserve a specific file. What do you think?

Reply:
I agree with that, thanks!